{"title":"Upload artifacts failed to Artifactory from AIX","slug":"2023/08/upload-artifacts-failed-on-aix","date":"2023-08-28T22:36:46.000Z","updated":"2025-08-03T20:33:27.010Z","comments":true,"path":"api/articles/2023/08/upload-artifacts-failed-on-aix.json","excerpt":"Recently my CI pipeline suddenly does not work on AIX 7.1 with error Caused by: javax.net.ssl.SSLHandshakeException: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath..","covers":null,"content":"<p>Recently my CI pipeline suddenly does not work on AIX 7.1 with error <code>Caused by: javax.net.ssl.SSLHandshakeException: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.</code>.</p>\n<span id=\"more\"></span>\n<details>\n<summary>Click to see more details about the failure log.</summary>\n\n<figure class=\"highlight bash\"><table><tr><td class=\"code\"><pre><span class=\"line\">22:13:30  Executing <span class=\"built_in\">command</span>: /bin/sh -c git <span class=\"built_in\">log</span> --pretty=format:%s -1</span><br><span class=\"line\">22:13:36  [consumer_0] Deploying artifact: https://artifactory.company.com/artifactory/generic-int-den/myproject/PRs/PR-880/1/myproject_bin_rel_AIX_5797b20.tar.Z</span><br><span class=\"line\">22:13:36  Error occurred <span class=\"keyword\">for</span> request GET /artifactory/api/system/version HTTP/1.1: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error.</span><br><span class=\"line\">22:13:36  Error occurred <span class=\"keyword\">for</span> request PUT /artifactory/generic-int-den/myproject/PRs/PR-880/1/cpplinter_bin_rel_AIX_5797b20.tar.Z;build.timestamp=1693273923987;build.name=PR-880;build.number=1 HTTP/1.1: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error.</span><br><span class=\"line\">22:13:36  [consumer_0] An exception occurred during execution:</span><br><span class=\"line\">22:13:36  java.lang.RuntimeException: javax.net.ssl.SSLHandshakeException: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.util.spec.SpecDeploymentConsumer.consumerRun(SpecDeploymentConsumer.java:44)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.producerConsumer.ConsumerRunnableBase.run(ConsumerRunnableBase.java:11)</span><br><span class=\"line\">22:13:36  \tat java.lang.Thread.run(Thread.java:785)</span><br><span class=\"line\">22:13:36  Caused by: javax.net.ssl.SSLHandshakeException: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.j.a(j.java:3)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.a(as.java:213)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.C.a(C.java:339)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.C.a(C.java:248)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.D.a(D.java:291)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.D.a(D.java:217)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.C.r(C.java:373)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.C.a(C.java:352)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.a(as.java:752)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.i(as.java:338)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.a(as.java:711)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.startHandshake(as.java:454)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.ServiceUnavailableRetryExec.execute(ServiceUnavailableRetryExec.java:85)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.client.PreemptiveHttpClient.execute(PreemptiveHttpClient.java:76)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.client.PreemptiveHttpClient.execute(PreemptiveHttpClient.java:64)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.client.JFrogHttpClient.sendRequest(JFrogHttpClient.java:133)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.client.JFrogService.execute(JFrogService.java:112)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.client.artifactory.services.Upload.execute(Upload.java:77)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.client.artifactory.ArtifactoryManager.upload(ArtifactoryManager.java:267)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.client.artifactory.ArtifactoryManager.upload(ArtifactoryManager.java:262)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.util.spec.SpecDeploymentConsumer.consumerRun(SpecDeploymentConsumer.java:39)</span><br><span class=\"line\">22:13:36  \t... 2 more</span><br><span class=\"line\">22:13:36  Caused by: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.util.f.a(f.java:107)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.util.f.b(f.java:143)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.util.e.a(e.java:6)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.aA.a(aA.java:99)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.aA.a(aA.java:112)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.aA.checkServerTrusted(aA.java:28)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.D.a(D.java:588)</span><br><span class=\"line\">22:13:36  \t... 29 more</span><br><span class=\"line\">22:13:36  Caused by: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.engineBuild(PKIXCertPathBuilderImpl.java:422)</span><br><span class=\"line\">22:13:36  \tat java.security.cert.CertPathBuilder.build(CertPathBuilder.java:268)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.util.f.a(f.java:120)</span><br><span class=\"line\">22:13:36  \t... 35 more</span><br><span class=\"line\">22:13:36  Caused by: java.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.BasicChecker.&lt;init&gt;(BasicChecker.java:111)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathValidatorImpl.engineValidate(PKIXCertPathValidatorImpl.java:199)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.myValidator(PKIXCertPathBuilderImpl.java:749)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.buildCertPath(PKIXCertPathBuilderImpl.java:661)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.buildCertPath(PKIXCertPathBuilderImpl.java:607)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.engineBuild(PKIXCertPathBuilderImpl.java:368)</span><br><span class=\"line\">22:13:36  \t... 37 more</span><br><span class=\"line\">22:13:36  Caused by: java.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.CertPathUtil.findIssuer(CertPathUtil.java:316)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.BasicChecker.&lt;init&gt;(BasicChecker.java:108)</span><br><span class=\"line\">22:13:36  \t... 42 more</span><br><span class=\"line\">22:13:36</span><br></pre></td></tr></table></figure>\n\n</details>\n\n<p>I have tried to download certificate.pem from my Artifactory and run this command, but the issue still there on my AIX 7.3.</p>\n<figure class=\"highlight bash\"><table><tr><td class=\"code\"><pre><span class=\"line\">/usr/java8_64/jre/bin/keytool -importcert -<span class=\"built_in\">alias</span> cacertalias -keystore /usr/java8_64/jre/lib/security/cacerts -file /path/to/your/certificate.pem</span><br></pre></td></tr></table></figure>\n\n<p>It investing it can not reproduce on my Windows, Linux and AIX 7.3 build machines.</p>\n<p>What’s the different between them? the only different is Java runtime. On my problematic AIX 7.1 build machine, I used a shared runtime which is a link to path <code>/tools/AIX-7.1/Java8_64-8.0.0.401/usr/java8_64/bin/java</code></p>\n<figure class=\"highlight bash\"><table><tr><td class=\"code\"><pre><span class=\"line\">bash-5.0$ /tools/AIX-7.1/Java8_64-8.0.0.401/usr/java8_64/bin/java -version</span><br><span class=\"line\">java version <span class=\"string\">&quot;1.8.0&quot;</span></span><br><span class=\"line\">Java(TM) SE Runtime Environment (build pap6480sr4fp1-20170215_01(SR4 FP1))</span><br><span class=\"line\">IBM J9 VM (build 2.8, JRE 1.8.0 AIX ppc64-64 Compressed References 20170209_336038 (JIT enabled, AOT enabled)</span><br><span class=\"line\">J9VM - R28_20170209_0201_B336038</span><br><span class=\"line\">JIT  - tr.r14.java.green_20170125_131456</span><br><span class=\"line\">GC   - R28_20170209_0201_B336038_CMPRSS</span><br><span class=\"line\">J9CL - 20170209_336038)</span><br><span class=\"line\">JCL - 20170215_01 based on Oracle jdk8u121-b13</span><br></pre></td></tr></table></figure>\n\n<p>And I have anther Java runtime installed my that machine which is <code>/usr/java8_64/bin/java</code></p>\n<figure class=\"highlight bash\"><table><tr><td class=\"code\"><pre><span class=\"line\">bash-5.0$ /usr/java8_64/bin/java -version</span><br><span class=\"line\">java version <span class=\"string\">&quot;1.8.0_241&quot;</span></span><br><span class=\"line\">Java(TM) SE Runtime Environment (build 8.0.6.5 - pap6480sr6fp5-20200111_02(SR6 FP5))</span><br><span class=\"line\">IBM J9 VM (build 2.9, JRE 1.8.0 AIX ppc64-64-Bit Compressed References 20200108_436782 (JIT enabled, AOT enabled)</span><br><span class=\"line\">OpenJ9   - 7d1059c</span><br><span class=\"line\">OMR      - d059105</span><br><span class=\"line\">IBM      - c8aee39)</span><br><span class=\"line\">JCL - 20200110_01 based on Oracle jdk8u241-b07</span><br></pre></td></tr></table></figure>\n\n<p>Actually the versions of these two java versions are different. I just changed configuration of <code>JavaPath</code> from <code>/tools/AIX-7.1/Java8_64-8.0.0.401/usr/java8_64/bin/java</code> to <code>/usr/java8_64/bin/java</code> in the Jenkins node and disconnect then launch agent again, it works.</p>\n<p>I don’t why it works, I don’t know much about Java certificate, if you know the reason please leave comments and let me know. Thank you.</p>\n<hr>\n<p>转载本站文章请注明作者和出处，请勿用于任何商业用途。欢迎关注公众号「DevOps攻城狮」</p>\n","more":"<details>\n<summary>Click to see more details about the failure log.</summary>\n\n<figure class=\"highlight bash\"><table><tr><td class=\"code\"><pre><span class=\"line\">22:13:30  Executing <span class=\"built_in\">command</span>: /bin/sh -c git <span class=\"built_in\">log</span> --pretty=format:%s -1</span><br><span class=\"line\">22:13:36  [consumer_0] Deploying artifact: https://artifactory.company.com/artifactory/generic-int-den/myproject/PRs/PR-880/1/myproject_bin_rel_AIX_5797b20.tar.Z</span><br><span class=\"line\">22:13:36  Error occurred <span class=\"keyword\">for</span> request GET /artifactory/api/system/version HTTP/1.1: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error.</span><br><span class=\"line\">22:13:36  Error occurred <span class=\"keyword\">for</span> request PUT /artifactory/generic-int-den/myproject/PRs/PR-880/1/cpplinter_bin_rel_AIX_5797b20.tar.Z;build.timestamp=1693273923987;build.name=PR-880;build.number=1 HTTP/1.1: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error.</span><br><span class=\"line\">22:13:36  [consumer_0] An exception occurred during execution:</span><br><span class=\"line\">22:13:36  java.lang.RuntimeException: javax.net.ssl.SSLHandshakeException: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.util.spec.SpecDeploymentConsumer.consumerRun(SpecDeploymentConsumer.java:44)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.producerConsumer.ConsumerRunnableBase.run(ConsumerRunnableBase.java:11)</span><br><span class=\"line\">22:13:36  \tat java.lang.Thread.run(Thread.java:785)</span><br><span class=\"line\">22:13:36  Caused by: javax.net.ssl.SSLHandshakeException: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.j.a(j.java:3)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.a(as.java:213)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.C.a(C.java:339)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.C.a(C.java:248)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.D.a(D.java:291)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.D.a(D.java:217)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.C.r(C.java:373)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.C.a(C.java:352)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.a(as.java:752)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.i(as.java:338)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.a(as.java:711)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.as.startHandshake(as.java:454)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.ServiceUnavailableRetryExec.execute(ServiceUnavailableRetryExec.java:85)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)</span><br><span class=\"line\">22:13:36  \tat org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.client.PreemptiveHttpClient.execute(PreemptiveHttpClient.java:76)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.client.PreemptiveHttpClient.execute(PreemptiveHttpClient.java:64)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.client.JFrogHttpClient.sendRequest(JFrogHttpClient.java:133)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.client.JFrogService.execute(JFrogService.java:112)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.client.artifactory.services.Upload.execute(Upload.java:77)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.client.artifactory.ArtifactoryManager.upload(ArtifactoryManager.java:267)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.client.artifactory.ArtifactoryManager.upload(ArtifactoryManager.java:262)</span><br><span class=\"line\">22:13:36  \tat org.jfrog.build.extractor.clientConfiguration.util.spec.SpecDeploymentConsumer.consumerRun(SpecDeploymentConsumer.java:39)</span><br><span class=\"line\">22:13:36  \t... 2 more</span><br><span class=\"line\">22:13:36  Caused by: com.ibm.jsse2.util.h: PKIX path building failed: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.util.f.a(f.java:107)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.util.f.b(f.java:143)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.util.e.a(e.java:6)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.aA.a(aA.java:99)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.aA.a(aA.java:112)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.aA.checkServerTrusted(aA.java:28)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.D.a(D.java:588)</span><br><span class=\"line\">22:13:36  \t... 29 more</span><br><span class=\"line\">22:13:36  Caused by: java.security.cert.CertPathBuilderException: PKIXCertPathBuilderImpl could not build a valid CertPath.; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.engineBuild(PKIXCertPathBuilderImpl.java:422)</span><br><span class=\"line\">22:13:36  \tat java.security.cert.CertPathBuilder.build(CertPathBuilder.java:268)</span><br><span class=\"line\">22:13:36  \tat com.ibm.jsse2.util.f.a(f.java:120)</span><br><span class=\"line\">22:13:36  \t... 35 more</span><br><span class=\"line\">22:13:36  Caused by: java.security.cert.CertPathValidatorException: The certificate issued by CN=DigiCert Global Root G2, OU=www.digicert.com, O=DigiCert Inc, C=US is not trusted; internal cause is:</span><br><span class=\"line\">22:13:36  \tjava.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.BasicChecker.&lt;init&gt;(BasicChecker.java:111)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathValidatorImpl.engineValidate(PKIXCertPathValidatorImpl.java:199)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.myValidator(PKIXCertPathBuilderImpl.java:749)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.buildCertPath(PKIXCertPathBuilderImpl.java:661)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.buildCertPath(PKIXCertPathBuilderImpl.java:607)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.PKIXCertPathBuilderImpl.engineBuild(PKIXCertPathBuilderImpl.java:368)</span><br><span class=\"line\">22:13:36  \t... 37 more</span><br><span class=\"line\">22:13:36  Caused by: java.security.cert.CertPathValidatorException: Certificate chaining error</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.CertPathUtil.findIssuer(CertPathUtil.java:316)</span><br><span class=\"line\">22:13:36  \tat com.ibm.security.cert.BasicChecker.&lt;init&gt;(BasicChecker.java:108)</span><br><span class=\"line\">22:13:36  \t... 42 more</span><br><span class=\"line\">22:13:36</span><br></pre></td></tr></table></figure>\n\n</details>\n\n<p>I have tried to download certificate.pem from my Artifactory and run this command, but the issue still there on my AIX 7.3.</p>\n<figure class=\"highlight bash\"><table><tr><td class=\"code\"><pre><span class=\"line\">/usr/java8_64/jre/bin/keytool -importcert -<span class=\"built_in\">alias</span> cacertalias -keystore /usr/java8_64/jre/lib/security/cacerts -file /path/to/your/certificate.pem</span><br></pre></td></tr></table></figure>\n\n<p>It investing it can not reproduce on my Windows, Linux and AIX 7.3 build machines.</p>\n<p>What’s the different between them? the only different is Java runtime. On my problematic AIX 7.1 build machine, I used a shared runtime which is a link to path <code>/tools/AIX-7.1/Java8_64-8.0.0.401/usr/java8_64/bin/java</code></p>\n<figure class=\"highlight bash\"><table><tr><td class=\"code\"><pre><span class=\"line\">bash-5.0$ /tools/AIX-7.1/Java8_64-8.0.0.401/usr/java8_64/bin/java -version</span><br><span class=\"line\">java version <span class=\"string\">&quot;1.8.0&quot;</span></span><br><span class=\"line\">Java(TM) SE Runtime Environment (build pap6480sr4fp1-20170215_01(SR4 FP1))</span><br><span class=\"line\">IBM J9 VM (build 2.8, JRE 1.8.0 AIX ppc64-64 Compressed References 20170209_336038 (JIT enabled, AOT enabled)</span><br><span class=\"line\">J9VM - R28_20170209_0201_B336038</span><br><span class=\"line\">JIT  - tr.r14.java.green_20170125_131456</span><br><span class=\"line\">GC   - R28_20170209_0201_B336038_CMPRSS</span><br><span class=\"line\">J9CL - 20170209_336038)</span><br><span class=\"line\">JCL - 20170215_01 based on Oracle jdk8u121-b13</span><br></pre></td></tr></table></figure>\n\n<p>And I have anther Java runtime installed my that machine which is <code>/usr/java8_64/bin/java</code></p>\n<figure class=\"highlight bash\"><table><tr><td class=\"code\"><pre><span class=\"line\">bash-5.0$ /usr/java8_64/bin/java -version</span><br><span class=\"line\">java version <span class=\"string\">&quot;1.8.0_241&quot;</span></span><br><span class=\"line\">Java(TM) SE Runtime Environment (build 8.0.6.5 - pap6480sr6fp5-20200111_02(SR6 FP5))</span><br><span class=\"line\">IBM J9 VM (build 2.9, JRE 1.8.0 AIX ppc64-64-Bit Compressed References 20200108_436782 (JIT enabled, AOT enabled)</span><br><span class=\"line\">OpenJ9   - 7d1059c</span><br><span class=\"line\">OMR      - d059105</span><br><span class=\"line\">IBM      - c8aee39)</span><br><span class=\"line\">JCL - 20200110_01 based on Oracle jdk8u241-b07</span><br></pre></td></tr></table></figure>\n\n<p>Actually the versions of these two java versions are different. I just changed configuration of <code>JavaPath</code> from <code>/tools/AIX-7.1/Java8_64-8.0.0.401/usr/java8_64/bin/java</code> to <code>/usr/java8_64/bin/java</code> in the Jenkins node and disconnect then launch agent again, it works.</p>\n<p>I don’t why it works, I don’t know much about Java certificate, if you know the reason please leave comments and let me know. Thank you.</p>\n<hr>\n<p>转载本站文章请注明作者和出处，请勿用于任何商业用途。欢迎关注公众号「DevOps攻城狮」</p>","categories":[{"name":"AIX","path":"api/categories/AIX.json"}],"tags":[{"name":"Troubleshooting","path":"api/tags/Troubleshooting.json"},{"name":"Artifactory","path":"api/tags/Artifactory.json"}]}